pwiz_tools/Skyline/Model/DocSettings/TransitionIonMobilityFiltering.cs

```diff
@@ -930,6 +930,12 @@ public IonMobilityFilter ChangeIonMobilityUnits(eIonMobilityUnits units)
 
         }
 
+        public IonMobilityFilter ChangeCollisionCrossSection(double ?ccs)
+        {
+            return Equals(CollisionalCrossSectionSqA, ccs) ? this : 
```

Comment:
I don't know about this check for equality. It doesn't seem like a bad idea per se, but we don't do it everywhere. So, it may just give a false sense of security that the change will be free of impact, when elements above it in the tree of objects may still get cloned and a new tree created.

And in this case, I don't see how it gets used with an equal value, since you are only using it to provide a value when you started with none.

You are not simply following a general style guideline in our code, but sort of make it a bit more confusing what our accepted best practice is. Should every ChangeProp() be doing this?

Reply:
The idea is just to avoid unneeded copying of things that aren't actually changing, there probably are a number of places where this would be a good thing to do.

It's true that the only current use of this newly created method is a case where the current value is none, but I didn't want to make that assumption in the implementation.